src/mockgyver.erl

```diff
@@ -1395,18 +1399,71 @@ mk_handle_undefined_function(Mod, ExportedFAs, FmtNoAction) ->
     %% each containing the inner case expression, though.
     func_from_str_fmt(
       "'$handle_undefined_function'(FnName, Args) ->
-           case lists:member({FnName, length(Args)}, ~p) of
+           Arity = length(Args),
+           case lists:member({FnName, Arity}, ~p) of
                true ->
                    case mockgyver:reg_call_and_get_action({~p,FnName,Args}) of
                        undefined ->
                            ~s;
                        ActionFun ->
-                           apply(ActionFun, Args)
+                           try
+                               apply(ActionFun, Args)
+                           catch
+                               Class:Reason:St0 ->
+                                   ToMF = {~p, FnName},
+                                   St = '$mockgyver_tweak_st'(
+                                            ActionFun, ToMF, Arity, St0),
+                                   erlang:raise(Class, Reason, St)
```

Comment:
There should be an error_handler:raise that takes care of the filtering of the '$handle_undefined_function'...